feat: add batches API with OpenAI compatibility

mattf opened the pull request:
Add complete batches API implementation with protocol, providers, and tests:

Core Infrastructure:
- Add batches API protocol using OpenAI Batch types directly
- Add Api.batches enum value and protocol mapping in resolver
- Add OpenAI "batch" file purpose support
- Include proper error handling (ConflictError, ResourceNotFoundError)

Reference Provider:
- Add ReferenceBatchesImpl with full CRUD operations (create, retrieve, cancel, list)
- Implement background batch processing with configurable concurrency
- Add SQLite KVStore backend for persistence
- Support /v1/chat/completions endpoint with request validation

Comprehensive Test Suite:
- Add unit tests for provider implementation with validation
- Add integration tests for end-to-end batch processing workflows
- Add error handling tests for validation, malformed inputs, and edge cases

Configuration:
- Add max_concurrent_batches and max_concurrent_requests_per_batch options
- Add provider documentation with sample configurations

Test with -

```
$ uv run llama stack build --image-type venv --providers inference=YOU_PICK,files=inline::localfs,batches=inline::reference --run &
$ LLAMA_STACK_CONFIG=http://localhost:8321 uv run pytest tests/unit/providers/batches tests/integration/batches --text-model YOU_PICK
```

addresses #3066

mattf commented:
@franciscojavierarceo ptal

ashwinb commented:
lgtm

I am not a fan of the "batches" name for an API but hey this time I can just ignore that piece and point to someone else :)

mattf commented:
> Should we note in the documentation somewhere that this API is under development?

i updated provider_codegen.py to include the api protocol's docstring in the providers docs and added a note about the development status.

fyi, this also picked up additional docs from agents, inference & eval.

mattf commented:
@ashwinb thoughts on the replay tests failing?

franciscojavierarceo commented on llama_stack/apis/batches/batches.py:
```diff
@@ -0,0 +1,99 @@
+        input_file_id: str,
+        endpoint: str,
+        completion_window: str,
+        metadata: dict[str, str] | None = None,
```
```suggestion
        metadata: dict[str, Any] | None = None,
```

mattf replied:
for compat, this definitely needs to be str

franciscojavierarceo commented on llama_stack/apis/batches/batches.py:
```diff
@@ -0,0 +1,99 @@
+        self,
+        input_file_id: str,
+        endpoint: str,
+        completion_window: str,
```
```suggestion
        completion_window: Literal["24h"],
```

mattf replied:
thanks, i missed this

mattf commented on llama_stack/providers/inline/batches/reference/batches.py:
```diff
@@ -0,0 +1,553 @@
+        batch_id = f"batch_{uuid.uuid4().hex[:16]}"
+        current_time = int(time.time())
+
+        batch = BatchObject(
```
:heart: idempotency is hugely valuable for apis like this.

since files are immutable, the params can serve as that id.

let me add an issue for that feature.

franciscojavierarceo replied:
>since files are immutable, the params can serve as that id.
100

franciscojavierarceo reviewed (approved):
🚀
